l2gersync/evm_downloader_sovereign.go

```diff
@@ -131,9 +134,21 @@ func (d *downloaderSovereign) buildAppender(
 			return fmt.Errorf("error parsing UpdateHashChainValue event log %+v: %w", l, err)
 		}
 
+		// Check if L1InfoTreeSync is up to date before trying to get GER info
+		ctx := context.Background()
```

Comment:
i dont think we need this check for uptodate
we can just check for mismatch, and if mismatches, check on l1 if ger exists

Reply:
Ok, I've just checked, `aggoracle` is inserting `GERs` on `L2` based on the last `L1 info tree leaf` from the `L1 Info Tree Syncer`, and since `L1 Info Tree Syncer` is only syncing `Finalized` blocks, I would say, this solution is then good enough in my opinion.